### PR TITLE
doc: improve doc of NewConstructor

### PR DESCRIPTION
**Pull request by ssiccha**
Constructors behave a bit differently under the method selection
than other operations but the documentation did not describe this.

### CODE REVIEW

**Comment by fingolfin** [comment]
While at it, maybe also hook up the `DeclareConstructor` documentation to the manual? I.e.
```diff
diff --git a/doc/ref/create.xml b/doc/ref/create.xml
index 224cccaa7..8a82fb7e5 100644
--- a/doc/ref/create.xml
+++ b/doc/ref/create.xml
@@ -1182,6 +1182,7 @@ See also Section <Ref Sect="More About Global Variables"/>.
 <#Include Label="DeclareProperty">
 <#Include Label="DeclareFilter">
 <#Include Label="DeclareOperation">
+<#Include Label="DeclareConstructor">
 <#Include Label="DeclareGlobalFunction">
 <#Include Label="DeclareGlobalVariable">
 <#Include Label="InstallValue">
```

File: lib/oper.g [hunk 1]
```diff
@@ -585,19 +585,52 @@ end );
 ##  <Ref Func="NewConstructor"/> returns a constructor <A>cons</A> with name
 ##  <A>name</A>.
 ##  The list <A>args-filts</A> describes requirements about the arguments
-##  of <A>constr</A>, namely the number of arguments must be equal to the length
-##  of <A>args-filts</A>, and the <M>i</M>-th argument must lie in the filter
-##  <A>args-filts</A><M>[i]</M>. Additionally a constructor expects the first
-##  argument to be a filter to then select a method to construct an object.
+##  of <A>cons</A>. Namely the number of arguments must be equal to the length
+##  of <A>args-filts</A>, and the <M>i</M>-th argument
```
Comment (fingolfin) [inline lib/oper.g]: Why this early wrapping? That causes an unnecessary diff (thus making it a little bit harder to review this PR), and I see no good reason for doing it?

File: lib/oper.g [hunk 2]
```diff
@@ -585,19 +585,52 @@ end );
...
+##  A constructor expects the first argument to be a <E>filter</E> instead
+##  of an object and it must be a subset of the filter
+##  <A>args-filts</A><M>[1]</M>.
 ##  <P/>
 ##  Each method that is installed for <A>cons</A> via
-##  <Ref Func="InstallMethod"/> must require that the <M>i</M>-th argument
-##  lies in the filter <A>args-filts</A><M>[i]</M>.
+##  <Ref Func="InstallMethod"/> must require that
+##  the <M>i</M>-th argument lies in the filter <A>args-filts</A><M>[i]</M>
+##  for <M>i \neq 1</M>.
```
Comment (fingolfin) [inline lib/oper.g]: Again, weird re-wrapping -- now I have to carefully check what you really changed :-(

File: lib/oper.g [hunk 3]
```diff
@@ -585,19 +585,52 @@ end );
...
+##  This has the following effects on the method selection for constructors.
+##  If <A>cons</A> is called with an argument tuple whose first argument is
+##  the filter <A>filt</A>, any method whose first argument is
+##  <E>more</E> specific than <A>filt</A> is applicable
+##  (if its other <A>args-filts</A> also match).
+##  Then the method with the <Q>most general</Q> filter <A>args-filts</A><M>[1]</M>
+##  is chosen, since the rank is computed by taking <M>-1</M> times the ranks
+##  of the involved filters.
+##  <P/>
+##  Imagine wanting to construct a new IsMatrixObject without caring
```
Comment (fingolfin) [inline lib/oper.g]: "Imagine" ? So, is this an example? Then I'd start with "For example" to make that clear.

Also, don't you mean `IsMatrixObj`?

File: lib/oper.g [hunk 4]
```diff
@@ -585,19 +585,52 @@ end );
...
+##  explicitly in which representation it is created.
+##  &GAP; then chooses a filter whis is <Q>as close</Q> to
+##  IsMatrixObject as possible.
```
Comment (fingolfin) [inline lib/oper.g]: This leaves me with more questions. E.g. why wouldn't I just use `IsMatrixObj`? Why would I only want to be "as close" as possible?
How about giving an actual example? Or perhaps declare a constructor, then install 2-3 methods, and document which of them is used for various (also given) example invocations?

File: lib/oper.g (hunk 2 above)
Comment (ssiccha) [inline lib/oper.g]: Ah, I just thought it would look cleaner in the end. Didn't think about the weird-looking diffs it would introduce. I'll keep that in mind in the future, thanks. :)

File: lib/oper.g (hunk 4 above)
Comment (stevelinton) [inline lib/oper.g]: I think "close" is slightly the wrong word. Method selection in this case will choose a constructor which guarantees to produce an object in IsMatrixObj. All other things being equal it will choose the one that promises the fewest extra filters on the grounds that that is presumably easiest to do.

Comment (ssiccha) [inline lib/oper.g]: Ok, I'll get on that,

File: lib/oper.g (hunk 2 above)
Comment (fingolfin) [inline lib/oper.g]: Luckily I can use http://www.jefftk.com/icdiff on my local computer to make it better, but the easier it is to already review in the browser, the faster you'll get meaningful feedback

**Comment by ssiccha** [comment]
The corresponding code is `INSTALL_METHOD_FLAGS` in `oper1.g:128`. We stumbled upon this when looking at the `NewMatrix` code.
